tools/reviewer/job/sync/DiffConverter.java

```diff
@@ -0,0 +1,173 @@
+          .setUpdatedAt(Instant.ofEpochMilli(diff.getModifiedTimestamp()).toString())
+          .setBaseBranchName("master")
+          .setHeadBranchName("D" + diff.getId())
+          .setRepo(githubPr.getRepo())
+          .addAllReviewComment(getReviewCommentsByRepoName(diff, githubPr))
+          .addAllIssueComment(getIssueComments(diff, githubPr))
+          .setOwner(githubPr.getOwner())
+          .setAssociatedReviewerDiff(diff.getId());
+      pullRequests.add(pullRequest.build());
+    }
+    return pullRequests;
+  }
+
+  private List<ReviewComment> getReviewCommentsByRepoName(Diff diff, GithubPr githubPr) {
+    List<ReviewComment> result = new ArrayList<>();
+    List<Thread> codeThreads =
+        diff.getCodeThreadList()
+            .stream()
+            .filter(thread -> thread.getRepoId().equals(githubPr.getRepo()))
+            .collect(Collectors.toList());
+
+    for (Thread thread : codeThreads) {
+      for (Comment comment : thread.getCommentList()) {
+        ReviewComment.Builder githubComment = ReviewComment.newBuilder();
+        githubComment
+            .setPath(thread.getFile().getFilename())
+            .setId(comment.getGithubCommentId())
+            .setPosition(
+                thread.getGithubCommentPosition() == 0
+                    ? getReviewCommentPosition(thread, githubPr)
+                    : thread.getGithubCommentPosition())
+            .setCommitId(thread.getFile().getCommitId())
+            .setUser(User.newBuilder().setEmail(comment.getCreatedBy()).build())
+            .setBody(comment.getContent())
+            .setCreatedAt(String.valueOf(Instant.ofEpochMilli(comment.getTimestamp())))
+            .setReviewerThreadId(thread.getId())
+            .build();
+        result.add(githubComment.build());
+      }
+    }
+    return result;
+  }
+
+  // TODO: Check if the state is correct
+  private String getPullRequestState(Diff.Status status) {
+    if (status.equals(Diff.Status.ACCEPTED)) {
+      return "close";
+    } else {
+      return "open";
+    }
+  }
+
+  private int getReviewCommentPosition(Thread thread, GithubPr githubPr) {
+    LineNumberConverter.Side commentSide =
+        thread.getFile().getCommitId().equals(thread.getCommitId())
+            ? LineNumberConverter.Side.RIGHT
+            : LineNumberConverter.Side.LEFT;
+    String patch = "";
+    try {
+      patch =
+          githubClient
+              .getCommit(
+                  CommitRequest.newBuilder()
+                      .setOwner(githubPr.getOwner())
+                      .setRepo(githubPr.getRepo())
+                      .setSha(thread.getFile().getCommitId())
+                      .build())
+              .getCommit()
+              .getFilesList()
+              .stream()
+              .filter(file -> thread.getFile().getFilename().equals(file.getFilename()))
+              .collect(Collectors.toList())
+              .get(0)
```

Comment:
How do we know to get position `0`? Do we assume there is only 1 element?

Reply:
Can you write in a comment why that assumption should be true?

Reply:
It should be true because of a filename (with path) is unique. 
I've added additional checking for a case if for some reason GitHub doesn't contain the commit for needed file.